src/main/java/citadel/Storage.java

```diff
@@ -0,0 +1,183 @@
+     * @return A {@link Deadline} task created from the input string.
+     * @throws CitadelException If the input string is invalid or incomplete.
+     */
+    private static Task createDeadline(String input) throws CitadelException {
+        Task t;
+        String[] words = input.split(" \\(by: ");
+        if (words.length < 2) {
+            throw new CitadelTaskNoInput();
+        }
+        String task = words[0].trim();
+        String deadline = words[1].substring(0, words[1].length() - 1).trim();
+        if (task.isEmpty() || deadline.isEmpty()) {
+            throw new CitadelTaskNoInput();
+        }
+
+        LocalDateTime deadlineFormatted = LocalDateTime.parse(deadline,
+                DateTimeFormatter.ofPattern("dd/MM/yyyy HH:mm"));
+        t = new Deadline(task, deadlineFormatted);
+        return t;
+    }
+
+    /**
+     * Creates an {@link Event} task from a string input.
+     * <p>
+     * The input string is parsed to extract the task description, start time, and end time.
+     * </p>
+     *
+     * @param input The string containing the task description, start time, and end time.
+     * @return An {@link Event} task created from the input string.
+     * @throws CitadelException If the input string is invalid or incomplete.
+     */
+    private static Task createEvent(String input) throws CitadelException {
+        Task t;
+        String[] words = input.split(" \\(from: ");
+        if (words.length < 2) {
+            throw new CitadelTaskNoInput();
+        }
+        String task = words[0].trim();
+        String[] timeline = words[1].split(" to: ");
+        if (timeline.length < 2) {
+            throw new CitadelTaskNoInput();
+        }
+        String from = timeline[0].trim();
+        String to = timeline[1].substring(0, timeline[1].length() - 1).trim();
+        if (task.isEmpty() || from.isEmpty() || to.isEmpty()) {
+            throw new CitadelTaskNoInput();
+        }
+        LocalDateTime fromFormatted = LocalDateTime.parse(from,
+                DateTimeFormatter.ofPattern("dd/MM/yyyy HH:mm"));
+        LocalDateTime toFormatted = LocalDateTime.parse(to,
+                DateTimeFormatter.ofPattern("dd/MM/yyyy HH:mm"));
+        t = new Event(task, fromFormatted, toFormatted);
+        return t;
+    }
+
+    /**
+     * Creates a {@link ToDo} task from a string input.
+     * <p>
+     * The input string is parsed to extract the task description.
+     * </p>
+     *
+     * @param input The string containing the task description.
+     * @return A {@link ToDo} task created from the input string.
+     * @throws CitadelException If the input string is invalid or incomplete.
+     */
+    private static Task createTodo(String input) throws CitadelException {
+        Task t;
+        String todo = input.trim();
+        if (todo.isEmpty()) {
+            throw new CitadelTaskNoInput();
+        }
+        t = new ToDo(todo);
```

Comment:
t is not very necessary here - could return directly since "t" does not provide extra description of your todo.